### PR TITLE
refactor: simplify the code

### PR DESCRIPTION
**Pull request by chenquan**

### CODE REVIEW

File: core/logx/writer.go [hunk 1]
```diff
@@ -269,7 +269,11 @@ func combineGlobalFields(fields []LogField) []LogField {
 		return fields
 	}
 
-	return append(globals.([]LogField), fields...)
```
Comment (chenquan) [inline core/logx/writer.go:272]: Ignoring the order, maybe this is ok? `append(fields, globals.([]LogField)...)`

**Review by kevwan — APPROVED** [review]
LGTM.

File: core/logx/writer.go (hunk 1 above)
Comment (kevwan) [reply core/logx/writer.go:272]: Fields are not order sensitive. It's ok.

Comment (chenquan) [reply core/logx/writer.go:272]: done.

File: core/logx/fields.go [hunk 2]
```diff
@@ -2,28 +2,23 @@ package logx
...
 	fieldsContextKey contextKey
 	globalFields     atomic.Value
-	globalFieldsLock sync.Mutex
 )
 
 type contextKey struct{}
 
 // AddGlobalFields adds global fields.
 func AddGlobalFields(fields ...LogField) {
-	globalFieldsLock.Lock()
```
Comment (kevwan) [inline core/logx/fields.go:19]: cannot delete this code, concurrent write may lose fields.

File: core/logx/fields.go [hunk 3]
```diff
@@ -2,28 +2,23 @@ package logx
...
-	defer globalFieldsLock.Unlock()
-
 	old := globalFields.Load()
 	if old == nil {
-		globalFields.Store(append([]LogField(nil), fields...))
```
Comment (kevwan) [inline core/logx/fields.go:24]: should create a new one.

**Review by kevwan — APPROVED** [review]

**Review by kevwan — APPROVED** [review]

**Pull request merged**